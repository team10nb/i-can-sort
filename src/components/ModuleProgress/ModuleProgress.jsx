/*
    Author: Shiliang Chen, Yuting Jiang

    Show the learning progress of an algorithm. 
*/

import React from "react";
import { makeStyles, withStyles} from "@material-ui/core/styles";
import LinearProgress from "@material-ui/core/LinearProgress";

export default function ModuleProgress(props) {

    const {width, color, progress, preOne} = props;
    const presentingWidth = preOne ? (width + 6) : (width);
    const presentingColor = progress === 100 ? color : color;

    // modify the default styles of linear progress bar
    const BorderLinearProgress = withStyles((theme) => ({
        root: {
            height: 10,
            borderTopLeftRadius: 0,
            borderTopRightRadius: 0,
            borderBottomRightRadius: 2,
            borderBottomLeftRadius: 2,
        },
        colorPrimary: {
            backgroundColor:
                theme.palette.grey[theme.palette.type === "light" ? 200 : 700],
        },
        bar: {
            borderRadius: 2,
            backgroundColor: presentingColor,
        },
    }))(LinearProgress);

    // control the width of the progress bar
    const useStyles = makeStyles({
        root: {
            flexGrow: 1,
            width: presentingWidth,
            minWidth: 200,
        },
    });
    
    const classes = useStyles();
    return (
        <div className={classes.root}>
            <BorderLinearProgress
                variant= "determinate" value= {progress}
            />
        </div>
    );
}
