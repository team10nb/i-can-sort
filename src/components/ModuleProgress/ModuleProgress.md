### **Developer**

Shiliang Chen, Yuting Jiang

Reference: MATERIAL-UI (https://material-ui.com)

###  **Introduction**

Used in conjunction with a Module component to indicate the learning progress of that Module.

###  **Import**

```html
import { makeStyles, withStyles} from "@material-ui/core/styles";
import LinearProgress from "@material-ui/core/LinearProgress";
```

###  **Props**
width, color, progress, preOne

| Name | Type | Default | Description |
| ---- | ---- | ------- | ----------- |
| width | number |         | The length of the progress bar. |
| color | string | #3f51b5 | The color of the progress bar. |
| progress| number | 100 | The progress of the progress bar, from 0 to 100. |
| preOne | bool |         | If true, increase the width by 6px. |


###  **Demo**

```jsx
<ModuleProgress width={500} color={'#1565c0'} progress={30} preOne = {false}/>
```