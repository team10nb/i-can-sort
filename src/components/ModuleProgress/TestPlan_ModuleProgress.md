## Test Plan
--------
<br>

#### Component: ModuleProgress

|Test Name|Description|Operation|Expected Outcome|
|----|------|--------|----------------|
| should match Snapshot | Compare new snapshot with old one. |  | Snapshots are the same. |



