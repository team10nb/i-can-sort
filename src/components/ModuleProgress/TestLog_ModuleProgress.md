reason/description里面写为什么test fail了
对**组件代码**做修改和修改的理由写在每个代码块前的注释里（由于没有通过test，而对**原码**做的修改）
有必要的话，贴一下test的代码
第一个version可以不写注释 最后一个version(pass的version)可以不写reason

#### ComponentName: ModuleProgress

|Test No.|Date|Pass/Fail|Failed test(s)|Reason/Description|Modified version|
|-------|----|---------|-------------------|----------------|----------------|
|1 | 1.22 | Fail | All | No code has been constructed |version 1|
|2 | 1.22 | Pass |/|Initial version is created|version 2|
|3 | 1.27 | Fail |Props passing changed|Change passing values|version 3|
|4 | 1.27 | Pass | /                     |Props updated|version 4|

##### Code - version 2
```javascript
import React from "react";
import { makeStyles, withStyles } from "@material-ui/core/styles";
import LinearProgress from "@material-ui/core/LinearProgress";

const BorderLinearProgress = withStyles((theme) => ({
    root: {
        boxShadow: "10x 10px 20px #888888",
        height: 10,
        borderTopLeftRadius: 2,
        borderTopRightRadius: 2,
        borderBottomRightRadius: 5,
        borderBottomLeftRadius: 5,
    },
    colorPrimary: {
        backgroundColor:
            theme.palette.grey[theme.palette.type === "light" ? 200 : 700],
    },
    bar: {
        borderRadius: 2,
        backgroundColor: "#1a90ff",
    },
}))(LinearProgress);

export default function ModuleProgress(props) {
    const useStyles = makeStyles({
        root: {
            flexGrow: 1,
            width: props.width,
            minWidth: 200,
        },
    });
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <BorderLinearProgress
                {...{ variant: "determinate", value: props.progress }}
            />
        </div>
    );
}

```

##### Code - version 3
```javascript
import React from "react";
import { makeStyles, withStyles} from "@material-ui/core/styles";
import LinearProgress from "@material-ui/core/LinearProgress";



export default function ModuleProgress(props) {

    const {width, color, progress, preOne} = props;
    const presentingWidth = preOne ? (width + 6) : (width);
    // TODO if complete??
    const presentingColor = progress === 100 ? color : color;

    // modify the default styles of linear progress bar
    const BorderLinearProgress = withStyles((theme) => ({
        root: {
            height: 10,
            borderTopLeftRadius: 0,
            borderTopRightRadius: 0,
            borderBottomRightRadius: 2,
            borderBottomLeftRadius: 2,
        },
        colorPrimary: {
            backgroundColor:
                theme.palette.grey[theme.palette.type === "light" ? 200 : 700],
        },
        bar: {
            borderRadius: 2,
            backgroundColor: presentingColor,
            // backgroundColor: "#1a90ff",
        },
    }))(LinearProgress);

    // control the width of the progress bar
    const useStyles = makeStyles({
        root: {
            flexGrow: 1,
            width: presentingWidth,
            minWidth: 200,
        },
    });
    
    const classes = useStyles();
    return (
        <div className={classes.root}>
            <BorderLinearProgress
                variant= "determinate" value= {progress}
            />
        </div>
    );
}



```

##### Code - version 4
```javascript
       import { render } from "@testing-library/react";
import { makeStyles, withStyles } from "@material-ui/core/styles";
import userEvent from "@testing-library/user-event";
import { screen } from "@testing-library/react";
import ModuleProgress from "./ModuleProgress";

xit("should be value of 15", () => {
    const borderLinearProgressSpy = jest
        .spyOn(ModuleProgress.prototype, "BorderLinearProgress")
        .mockImplementation(jest.fn());

    const props = { width: 200, progress: 10 };
    const content = render(<ModuleProgress {...props} />);
    expect(borderLinearProgressSpy).toBeCalledTimes(1);
    expect(borderLinearProgressSpy).toBeCalledWith({
        variant: "determinate",
        value: 10,
    });
});

```


