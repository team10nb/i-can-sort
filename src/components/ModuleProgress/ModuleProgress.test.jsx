/*
    Author: Shiliang Chen, Yuting Jiang
*/

import { render } from "@testing-library/react";
import ModuleProgress from "./ModuleProgress";

test("should match Snapshot", () => {


    const props = { width: 200, progress: 10 };
    const content = render(<ModuleProgress {...props} />);
    expect(content).toMatchSnapshot();
});
