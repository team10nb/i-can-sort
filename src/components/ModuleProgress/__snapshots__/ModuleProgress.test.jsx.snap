// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="10"
          class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-2 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-3 MuiLinearProgress-determinate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-4 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
            style="transform: translateX(-90%);"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-2 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-3 MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-4 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style="transform: translateX(-90%);"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
